core/src/main/java/com/airbnb/aerosolve/core/models/AdditiveModel.java

```diff
@@ -183,9 +249,14 @@ public void addFunction(String featureFamily, String featureName,
     }
   }
 
+  public void addDenseFunction(String featureName, NDTreeModel ndtreeModel) {
+    MultiDimensionSpline spline = new MultiDimensionSpline(ndtreeModel);
+    Map<String, Function> dense = getOrCreateDenseWeights();
+    dense.put(featureName, spline);
+  }
+
   // Update weights based on gradient and learning rate
-  public void update(float grad,
-                     float learningRate,
+  public void update(float delta,
```

Comment:
instead of `delta`, if the input is `-delta` (same as the original implementation), it will be corresponding to gradient, I feel the code readability will be better in that way